datasets/dataset_cifar10.py

```diff
@@ -27,13 +33,35 @@ def preprocess_dataset_inputs(x):
 
 # Data samples pre-processing method for labels
 def preprocess_dataset_labels(y):
-    y = keras.utils.to_categorical(y, num_classes)
+    y = to_categorical(y, num_classes)
 
     return y
 
 
 def generate_new_dataset():
-    (x_train, y_train), (x_test, y_test) = cifar10.load_data()
+    attempts = 0
+    while True:
+        try:
+            (x_train, y_train), (x_test, y_test) = cifar10.load_data()
+            break
+        except HTTPError as e:
+            logger.debug(
+                f'URL fetch failure on https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz : {e.code} -- {e.msg}')
+            if attempts < constants.NUMBER_OF_DOWNLOAD_ATTEMPTS:
+                sleep(2)
+                attempts += 1
+            else:
+                raise
+        except URLError as e:
+            logger.debug(
+                f'URL fetch failure on https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz : '
+                f'{e.errno} -- {e.reason}')
```

Comment:
Why do we we log `{e.code} -- {e.msg}` for `HTTPError `and `{e.errno} -- {e.reason}` for the `URLError `?

Reply:
In fact .msg and .reason are the same things for HTTPerror but msg does not exist for URLlib. The URLlib does not hold .code

Reply:
the .code will be for instance the famous 404, which I think is useful to log .